examples/demo13.py

```diff
@@ -0,0 +1,378 @@
+        # about WCS:
+        final_image = galsim.ImageF(wfirst.n_pix,wfirst.n_pix, wcs=wcs)
+
+        # Draw the galaxies into the image.
+        for i_gal in xrange(n_use):
+            logger.info('Drawing image for the object at row %d in the input catalog'%i_gal)
+
+            # We want to only draw the galaxy once (for speed), not over and over with different
+            # sub-pixel offsets.  For this reason we will ignore the sub-pixel offset entirely.
+            stamp = galsim.Image(stamp_size, stamp_size)
+            gal_list[i_gal].drawImage(filter_, image=stamp)
+
+            # Have to find where to place it:
+            for i_gal_use in range(i_gal*n_per_gal, i_gal*(n_per_gal+1)):
+                # Account for the fractional part of the position:
+                ix = int(math.floor(x_stamp[i_gal_use]+0.5))
+                iy = int(math.floor(y_stamp[i_gal_use]+0.5))
+                # We're not going to actually use this offset.
+                offset = galsim.PositionD(x_stamp[i_gal]-ix, y_stamp[i_gal]-iy)
+
+                # Create a nominal bound for the postage stamp given the integer part of the
+                # position.
+                stamp_bounds = galsim.BoundsI(ix-0.5*stamp_size, ix+0.5*stamp_size-1, 
+                                              iy-0.5*stamp_size, iy+0.5*stamp_size-1)
+                stamp.setOrigin(galsim.PositionI(stamp_bounds.xmin, stamp_bounds.ymin))
+
+                # Find the overlapping bounds between the large image and the individual postage
+                # stamp.
+                bounds = stamp_bounds & final_image.bounds
+
+                # Copy the image into the right place in the big image.  There is a difference in
+                # normalization between COSMOS images and these that we account for using a
+                # numerical factor.
+                final_image[bounds] += stamp[bounds]/0.03**2
+
+        # Now we're done with the per-galaxy drawing for this image.  The rest will be done for the
+        # entire image at once.
+        logger.info('Postage stamps of all galaxies drawn on a single big image for this filter.')
+        logger.info('Adding the sky level, noise and detector non-idealities.')
+
+        # First we get the amount of zodaical light for a position corresponding to the center of
+        # this SCA.  Since we have supplied an exposure time, the results will be returned to us in
+        # e-/pix.  Then we multiply this by a factor >1 to account for the amount of stray light
+        # that is expected.  Technically one should make a position-dependent sky level, but this
+        # will be a fairly flat function of position, so using a constant is not too bad.
+        sky_level_pix = wfirst.getSkyLevel(filters[filter_name], position=SCA_cent_pos,
+                                           exp_time=wfirst.exptime)
+        sky_level_pix *= (1.0 + wfirst.stray_light_fraction)
```

Comment:
At this point, you should use the `wcs.makeSkyImage` function
```
sky_image = wcs.makeSkyImage(final_image.copy(), sky_level)
sky_image += wfirst.thermal_backgrounds...
final_image += sky_image
```
where `sky_level` is my recommended change to the return value of `wfirst.getSkyLevel` in e-/arcsec**2.